spec/models/manager_refresh/inventory_collection_default_spec.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+describe ManagerRefresh::InventoryCollectionDefault do
+  before { allow(MiqServer).to receive(:my_zone) { 'My zone' } }
+  let(:zone) { Zone.first }
+
+  context 'Amazon' do
+    before do
+      ems = new_amazon_ems
+      with_vcr_data { expect { EmsRefresh.refresh(ems) }.to change { VmOrTemplate.count }.from(0) }
+      expect { ems.destroy }.to_not change { VmOrTemplate.count }
+      expect(ExtManagementSystem.count).to be_zero
+    end
+
+    it 'pickups existing Vm records on EMS re-adding' do
+      expect(Vm.count).to_not be_zero
+      with_vcr_data { expect { EmsRefresh.refresh(new_amazon_ems) }.to_not change { Vm.count } }
+    end
+
+    it 'pickups existing Template records on EMS re-adding' do
+      expect(MiqTemplate.count).to_not be_zero
+      with_vcr_data { expect { EmsRefresh.refresh(new_amazon_ems) }.to_not change { MiqTemplate.count } }
+    end
+  end
+
+  private
+
+  def new_amazon_ems
+    FactoryGirl.create(:ems_amazon_with_vcr_authentication, :zone => zone, :provider_region => 'eu-central-1')
```

Comment:
zone is already part of the factory https://github.com/Ladas/manageiq-providers-amazon/blob/32eb26e08f5001fd319d31e4903eb681e2326d59/spec/factories/ext_management_system.rb#L10

so if you remove from here, the specs should pass

Reply:
yep, its some leftover from the preliminary research